softgroup/frontend/rest-server/src/main/java/com/softgroup/server/rest/service/RestTool.java

```diff
@@ -0,0 +1,42 @@
+package com.softgroup.server.rest.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.softgroup.common.datamapper.DataMapper;
+import com.softgroup.common.protocol.Request;
+import com.softgroup.common.protocol.RoutingData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import com.softgroup.token.api.JwtUserIdentifier;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+/**
+ * Created by user on 07.04.2017.
+ */
+@Service
+public class RestTool implements RestToolService{
+
+    @Autowired
+    private DataMapper dataMapper;
+
+    public Request<LinkedHashMap> parseRequestFromJson(String json){
+        Request<LinkedHashMap> map = dataMapper.mapData(json, new TypeReference<Request<HashMap>>() {
+        });
+        return map;
+    }
+
+    public Request<LinkedHashMap> setRoutingData(Request<LinkedHashMap> request){
```

Comment:
@Kun-Kun Return new request in this method or don't return anything. Mutating input parameter and returning something in one method is very bad practice.